### ISSUE TITLE
Search bar border leaks background color

### ISSUE THREAD

**Issue by paul-hansen** [rust-lang/crates.io#8677]
### Current Behavior

Minor thing, when you focus the search bar at the top of any crates.io page, it adds a yellow border to the search box. This border is ever so slightly separated from the search icon button on the right side and causes the background color to bleed through:
![image](https://github.com/rust-lang/crates.io/assets/7019130/23678269-878e-4dfb-bf82-e8c1f58a22aa)
This happens on the light theme too but it is more obvious with the dark theme since the background has more contrast.

### Expected Behavior

Expected the search button/icon to blend seamlessly with the yellow border since they are both the same color. If this gap was intentional it should probably be larger so it doesn't look like a mistake/graphical glitch.

### Environment

- Browser: Firefox 126.0
- OS: EndeavorOS (Arch based rolling distro)

### Anything else?

Thanks for the dark theme, love it!

**Comment by Turbo87** [comment]
nope, it's not intentional. appears to be a bug in Firefox though, the other browsers are rendering it correctly :-/

**Comment by paul-hansen** [comment]
I do see it on Chrome for Android too:
![Screenshot_20240519_040642](https://github.com/rust-lang/crates.io/assets/7019130/9b0d3d5f-35d6-4c44-8d91-1ccb622ddf6c)
(smoother looking because phone has higher dpi)

**Comment by JustLinuxUser** [comment]
Still having this issue (Firefox 134 beta). Some resolutions are worse then others ![Image](https://github.com/user-attachments/assets/d8a668ff-7888-40d9-9006-afab8b3d0779)
